src/main/java/com/mparticle/kits/AppboyKit.java

```diff
@@ -222,12 +229,28 @@ public List<ReportingMessage> logEvent(CommerceEvent event) {
         }
         List<MPEvent> eventList = CommerceEventUtils.expand(event);
         if (eventList != null) {
-            for (int i = 0; i < eventList.size(); i++) {
+            if (expandNonPurchaseCommerceEvents) {
+                for (int i = 0; i < eventList.size(); i++) {
+                    try {
+                        logEvent(eventList.get(i));
+                        messages.add(ReportingMessage.fromEvent(this, event));
+                    } catch (Exception e) {
+                        Logger.warning("Failed to call logCustomEvent to Braze kit: " + e.toString());
+                    }
+                }
+            } else {
+                JSONArray productArray = new JSONArray();
+                for (int i = 0; i < eventList.size(); i++) {
+                    Map<String, Object> newAttributes = eventList.get(i).getCustomAttributes();
+                    newAttributes.put("custom attributes", event.getCustomAttributes());
+                    productArray.put(newAttributes);
+                }
                 try {
-                    logEvent(eventList.get(i));
+                    BrazeProperties brazeProperties = new BrazeProperties(new JSONObject().put("Transaction ID", event.getTransactionAttributes().getId()).put("products", productArray));
```

Comment:
`event.getTransactionAttributes()` is Nullable :/

```suggestion
                    JSONObject json = new JSONObject().put("products", productArray);
                    TransactionAttributes transactionAttributes = event.getTransactionAttributes();
                    if (transactionAttributes != null) {
                        json.put("Transaction ID",  transactionAttributes.getId();
                    }
                    BrazeProperties brazeProperties = new BrazeProperties(json);
```